cloudinit/config/cloud-init-schema.json

```
@@ -543,6 +543,244 @@
+            },
+            "package_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to upgrade packages. Happens before install. Default: ``false``"
+            },
+            "package_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to reboot the system if required. Default: ``false``"
+            },
+            "apt_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_update``. Default: ``false``"
+            },
+            "apt_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_upgrade``. Default: ``false``"
+            },
+            "apt_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_reboot_if_required``. Default: ``false``"
+            }
+        }
+    },
+    "cc_phone_home": {
+      "type": "object",
+      "properties": {
+        "phone_home": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["url"],
+          "properties": {
+            "url": {
+              "type": "string",
+              "description": "The URL to send the phone home data to."
+            },
+            "post": {
+              "description": "A list of keys to post or ``all``. Default: ``all``",
+              "oneOf": [
+                {"enum": ["all"]},
+                {
+                  "type": "array",
+                  "items": {
+                    "type": "string",
+                    "enum": [
+                      "pub_key_dsa",
+                      "pub_key_rsa",
+                      "pub_key_ecdsa",
+                      "pub_key_ed25519",
+                      "instance_id",
+                      "hostname",
+                      "fqdn"
+                    ]
+                  }
+                }
+              ]
+            },
+            "tries": {
+              "type": "integer",
+              "description": "The number of times to try sending the phone home data. Default: ``10``",
+              "default": 10
+            }
+          }
+        }
+      }
+    },
+    "cc_power_state_change": {
+      "type": "object",
+      "properties": {
+        "power_state": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["mode"],
+          "properties": {
+            "delay": {
+              "description": "Time in minutes to delay after cloud-init has finished. Can be ``now`` or an integer specifying the number of minutes to delay. Default: ``now``",
+              "default": "now",
+              "oneOf": [
+                {"type": "integer"},
```

Let's put a "minimum": 0 here too
```suggestion
                {"type": "integer", "minimum": 0},
```